datadog_checks_base/datadog_checks/base/checks/openmetrics/v2/transform.py

```diff
@@ -0,0 +1,173 @@
+        )
+        self.non_cumulative_histogram_buckets = self.histogram_buckets_as_distributions or is_affirmative(
+            config.get('non_cumulative_histogram_buckets', False)
+        )
+
+        # Accessible to every transformer
+        self.global_options = {
+            'collect_histogram_buckets': self.collect_histogram_buckets,
+            'histogram_buckets_as_distributions': self.histogram_buckets_as_distributions,
+            'non_cumulative_histogram_buckets': self.non_cumulative_histogram_buckets,
+        }
+
+        metrics_config = deepcopy(self.normalize_metric_config(config))
+
+        self.transformer_data = {}
+        self.metric_patterns = []
+        for raw_metric_name, config in metrics_config.items():
+            escaped_metric_name = re.escape(raw_metric_name)
+
+            if raw_metric_name != escaped_metric_name:
+                config.pop('name')
+                self.metric_patterns.append((re.compile(raw_metric_name), config))
+            else:
+                try:
+                    self.transformer_data[raw_metric_name] = self.compile_transformer(config)
+                except Exception as e:
+                    error = f'Error compiling transformer for metric `{raw_metric_name}`: {e}'
+                    raise_from(type(e)(error), None)
+
+    def get(self, metric):
+        metric_name = metric.name
+
+        transformer_data = self.transformer_data.get(metric_name)
+        if transformer_data is not None:
+            metric_type, transformer = transformer_data
+            if metric_type == DEFAULT_METRIC_TYPE and self.skip_native_metric(metric):
+                return
+
+            return transformer
+        elif self.metric_patterns:
+            for metric_pattern, config in self.metric_patterns:
+                if metric_pattern.search(metric_name):
+                    metric_type, transformer = self.compile_transformer({'name': metric_name, **config})
+                    if self.cache_metric_wildcards:
+                        self.transformer_data[metric_name] = metric_type, transformer
+
+                    if metric_type == DEFAULT_METRIC_TYPE and self.skip_native_metric(metric):
+                        return
+
+                    return transformer
+
+        self.logger.debug('Skipping metric `%s` as it is not defined in `metrics`', metric_name)
+
+    def compile_transformer(self, config):
+        metric_name = config.pop('name')
+        if not isinstance(metric_name, str):
+            raise TypeError('field `name` must be a string')
+
+        metric_type = config.pop('type')
+        if not isinstance(metric_type, str):
+            raise TypeError('field `type` must be a string')
+
+        factory = TRANSFORMERS.get(metric_type)
+        if factory is None:
+            raise ValueError(f'unknown type `{metric_type}`')
+
+        return metric_type, factory(self.check, metric_name, config, self.global_options)
+
+    def skip_native_metric(self, metric):
```

Comment:
Should this be named `skip_nonnative_metric`?  How would this case happen if `metric_type` != `metric.type`?

Reply:
This is called for metrics with _transformer_ type `native` (the default) which uses whatever the endpoint says

Reply:
Ok - so this protects against metrics which self-describe as a native type which don't actually support, as described here: https://github.com/OpenObservability/OpenMetrics/blob/master/specification/OpenMetrics.md#metric-types

Can we add a comment to that effect to clarify?